plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/AI.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.ai;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.HttpException;
+import org.jkiss.dbeaver.model.ai.completion.*;
+import org.jkiss.dbeaver.model.ai.utils.AIUtils;
+import org.jkiss.dbeaver.model.ai.utils.ThrowableSupplier;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.sql.SQLUtils;
+
+import java.util.List;
+import java.util.concurrent.Flow;
+import java.util.stream.Stream;
+
+public class AI {
+    public static final AI INSTANCE = new AI();
```

Comment:
It doesn't make sense to have this singleton. It has no state. You can make all functions static.

Generally I don't like this approach, 
- there is no session concept
- engine cannot be specified for a particular request
- all settings/tokens are global and static